### PR TITLE
fix(install_scylla): adding workaround to JDK

### PR DESCRIPTION
**Pull request by fgelcer**
because of issue https://github.com/scylladb/scylla/issues/10442
we need to add here this workaround, as the
base versions patches don't include the fix
for this issue.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2205,6 +2205,19 @@ def install_scylla(self, scylla_repo):
...
+        # hence this is a temporary workaround to make the rolling upgrade tests to pass, until the latest
+        # patch of the supported releases will include the fix.
+        package_manager = 'yum' if self.is_rhel_like() else 'apt'
+        self.remoter.sudo(f'{package_manager} install zip unzip -y')
+        self.remoter.run('curl -s "https://get.sdkman.io" | bash')
+        command = dedent(f"""
+            source "$HOME/.sdkman/bin/sdkman-init.sh"
+            sdk install java 8.0.302-open
+        """)
+        self.remoter.run(f'bash -cxe "{command}"')
```
Comment (fruch) [inline sdcm/cluster.py:2219]: nitpick, you can use `shell_script_cmd` helper, it doesn't the dedent and the bash prefix for you.

File: sdcm/cluster.py [hunk 2]
```diff
@@ -2205,6 +2205,19 @@ def install_scylla(self, scylla_repo):
                 'sudo apt-get install -y -o Dpkg::Options::="--force-overwrite" -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" '
                 '--allow-unauthenticated {}'.format(self.scylla_pkg()))
 
+        # THIS IS A WORKAROUND FOR ISSUE https://github.com/scylladb/scylla/issues/10442
+        # the issue is related to JDK version, and the fix was added to later patches of multiple base versions,
+        # hence this is a temporary workaround to make the rolling upgrade tests to pass, until the latest
+        # patch of the supported releases will include the fix.
+        package_manager = 'yum' if self.is_rhel_like() else 'apt'
```
Comment (fruch) [inline sdcm/cluster.py:2212]: maybe we should first check which java version is installed as default ? before we apply this patch/hack ?

**Comment by fgelcer** [comment]
@fruch [this build](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fabio/job/2021.1/job/rolling-upgrade-centos7-test-jdk-workaround/6/) passed the setup phase.
ready to be merged, then

**Comment by fgelcer** [comment]
also @fruch , this PR i'm sending to fix `branch-2021.1`
do you need this fix in other branches as well?

File: sdcm/cluster.py (hunk 2 above)
Comment (fgelcer) [reply sdcm/cluster.py:2212]: this specific trick is to make 2021.1 upgrade tests to pass, as all the base versions will either not receive the fix (as they are already deprecated), or the fix didn't yet reach a released patch...
we can do it for `master` or other branches, and in that case, we do need to check the installed version first, as we do have a fix already to most versions, so it is not needed

**Comment by fruch** [comment]
> also @fruch , this PR i'm sending to fix `branch-2021.1` do you need this fix in other branches as well?

I was failing for provision test on GCE, since we don't yet have release with the fix.
We'll wait for a release

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fgelcer** [comment]
> I was failing for provision test on GCE, since we don't yet have release with the fix.
> We'll wait for a release
>

in that case, i can send a PR to master, and in case the version will be either 4.5 or 4.6, then use this hack to allow it to pass... WDYT?

**Comment by fgelcer** [comment]
@fruch , i see it is still complaining about the same error, but in a later stage:
```
nodetool: Failed to connect to '127.0.0.1:7199' - URISyntaxException: 'Malformed IPv6 address at index 7: rmi://[127.0.0.1]:7199'
```

**Comment by fruch** [comment]
> @fruch , i see it is still complaining about the same error, but in a later stage:
> 
> ```
> nodetool: Failed to connect to '127.0.0.1:7199' - URISyntaxException: 'Malformed IPv6 address at index 7: rmi://[127.0.0.1]:7199'
> ```

are you such this sdkman is setting the default java ? or it's an extra command you need todo ?

looking at https://sdkman.io/usage, I thing setting the configuration option `sdkman_auto_answer` might be needed

**Comment by fgelcer** [comment]
> > @fruch , i see it is still complaining about the same error, but in a later stage:
> > ```
> > nodetool: Failed to connect to '127.0.0.1:7199' - URISyntaxException: 'Malformed IPv6 address at index 7: rmi://[127.0.0.1]:7199'
> > ```
> 
> are you such this sdkman is setting the default java ? or it's an extra command you need todo ?
> 
> looking at https://sdkman.io/usage, I thing setting the configuration option `sdkman_auto_answer` might be needed

i saw it sets the java to defaults to the one being installed, but what can be the case, is that once it is installed, we reset the `alternative` java to the one previously installed...
i will try to skip it, or even remove it for this branch

**Comment by fruch** [comment]
> > > @fruch , i see it is still complaining about the same error, but in a later stage:
> > > ```
> > > nodetool: Failed to connect to '127.0.0.1:7199' - URISyntaxException: 'Malformed IPv6 address at index 7: rmi://[127.0.0.1]:7199'
> > > ```
> > 
> > 
> > are you such this sdkman is setting the default java ? or it's an extra command you need todo ?
> > looking at https://sdkman.io/usage, I thing setting the configuration option `sdkman_auto_answer` might be needed
> 
> i saw it sets the java to defaults to the one being installed, but what can be the case, is that once it is installed, we reset the `alternative` java to the one previously installed... i will try to skip it, or even remove it for this branch

I think doing this before calling `sdk install`, might do the trick:
```
sed -i s/sdkman_auto_answer=false/sdkman_auto_answer=true/  ~/.sdkman/etc/config
```

**Pull request merged**